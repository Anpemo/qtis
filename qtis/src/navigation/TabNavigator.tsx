import React from 'react'
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs'
import { Browser, Profile, TabBar } from '../screens'

const Tab = createBottomTabNavigator()

function TabNavigator () {
  return (
      <Tab.Navigator tabBar ={props => <TabBar {...props} />}>
          <Tab.Screen name='Browser' component={Browser} initialParams={{ icon: 'search1' }}/>
          <Tab.Screen name='Profile' component={Profile} initialParams={{ icon: 'user' }}/>
      </Tab.Navigator>
  )
}

export default TabNavigator
