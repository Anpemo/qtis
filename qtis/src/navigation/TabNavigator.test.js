import React from 'react'
import TabNavigator from './TabNavigator'
import { render } from '@testing-library/react-native'
import { NavigationContainer } from '@react-navigation/native'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

describe('Given a TabNavigator', () => {
  let component
  beforeEach(() => {
    const mockStore = configureStore()
    component = (
        <Provider store = {
          mockStore()
        }>

        <NavigationContainer>
            <TabNavigator />
        </NavigationContainer>
        </Provider>
    )
  })
  describe('When calling it', () => {
    test('Then it will match the snapshot', () => {
      const tabNavigator = render(component)
      expect(tabNavigator).toMatchSnapshot()
    })
  })
})
