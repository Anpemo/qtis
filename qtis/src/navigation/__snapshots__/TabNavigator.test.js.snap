// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a TabNavigator When calling it Then it will match the snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="auto"
        removeClippedSubviews={false}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <RNCSafeAreaView
            style={
              Object {
                "flex": 1,
                "margin": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#141414",
                  "fontSize": 25,
                  "margin": 10,
                  "padding": 5,
                }
              }
            >
              CHOOSE A CATEGORY
            </Text>
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <RCTScrollView
                data={
                  Array [
                    Object {
                      "id": 1,
                      "name": "soaps",
                      "src": 1,
                    },
                    Object {
                      "id": 2,
                      "name": "creams",
                      "src": 1,
                    },
                    Object {
                      "id": 3,
                      "name": "masks",
                      "src": 1,
                    },
                    Object {
                      "id": 4,
                      "name": "oils",
                      "src": 1,
                    },
                    Object {
                      "id": 5,
                      "name": "tools",
                      "src": 1,
                    },
                    Object {
                      "id": 6,
                      "name": "shampoos",
                      "src": 1,
                    },
                  ]
                }
                disableVirtualization={false}
                getItem={[Function]}
                getItemCount={[Function]}
                horizontal={false}
                initialNumToRender={10}
                keyExtractor={[Function]}
                maxToRenderPerBatch={10}
                onContentSizeChange={[Function]}
                onEndReachedThreshold={2}
                onLayout={[Function]}
                onMomentumScrollEnd={[Function]}
                onScroll={[Function]}
                onScrollBeginDrag={[Function]}
                onScrollEndDrag={[Function]}
                removeClippedSubviews={false}
                renderItem={[Function]}
                scrollEventThrottle={50}
                stickyHeaderIndices={Array []}
                testID="flatlistTest"
                updateCellsBatchingPeriod={50}
                viewabilityConfigCallbackPairs={Array []}
                windowSize={21}
              >
                <View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      style={
                        Object {
                          "flexDirection": "row",
                        }
                      }
                    >
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "borderRadius": 8,
                            "elevation": 4,
                            "height": 466.9,
                            "margin": 10,
                            "opacity": 1,
                            "shadowColor": "#000",
                            "shadowOffset": Object {
                              "height": 2,
                              "width": 0,
                            },
                            "shadowOpacity": 0.23,
                            "shadowRadius": 2.62,
                            "width": 322.5,
                          }
                        }
                        testID={1}
                      >
                        <View>
                          <Image
                            source={1}
                            style={
                              Object {
                                "borderRadius": 8,
                                "height": "100%",
                                "resizeMode": "cover",
                                "width": "100%",
                              }
                            }
                          />
                        </View>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "rgba(255, 255, 255, 0.75)",
                              "borderRadius": 8,
                              "bottom": 0,
                              "flex": 1,
                              "height": "50%",
                              "justifyContent": "flex-end",
                              "position": "absolute",
                              "width": "100%",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "alignSelf": "center",
                                "color": "#D4A373",
                                "fontFamily": "MontserratBold",
                                "fontSize": 25,
                                "justifyContent": "flex-end",
                                "marginBottom": 10,
                              }
                            }
                          >
                            SOAPS
                          </Text>
                        </View>
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "borderRadius": 8,
                            "elevation": 4,
                            "height": 466.9,
                            "margin": 10,
                            "opacity": 1,
                            "shadowColor": "#000",
                            "shadowOffset": Object {
                              "height": 2,
                              "width": 0,
                            },
                            "shadowOpacity": 0.23,
                            "shadowRadius": 2.62,
                            "width": 322.5,
                          }
                        }
                        testID={2}
                      >
                        <View>
                          <Image
                            source={1}
                            style={
                              Object {
                                "borderRadius": 8,
                                "height": "100%",
                                "resizeMode": "cover",
                                "width": "100%",
                              }
                            }
                          />
                        </View>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "rgba(255, 255, 255, 0.75)",
                              "borderRadius": 8,
                              "bottom": 0,
                              "flex": 1,
                              "height": "50%",
                              "justifyContent": "flex-end",
                              "position": "absolute",
                              "width": "100%",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "alignSelf": "center",
                                "color": "#D4A373",
                                "fontFamily": "MontserratBold",
                                "fontSize": 25,
                                "justifyContent": "flex-end",
                                "marginBottom": 10,
                              }
                            }
                          >
                            CREAMS
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      style={
                        Object {
                          "flexDirection": "row",
                        }
                      }
                    >
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "borderRadius": 8,
                            "elevation": 4,
                            "height": 466.9,
                            "margin": 10,
                            "opacity": 1,
                            "shadowColor": "#000",
                            "shadowOffset": Object {
                              "height": 2,
                              "width": 0,
                            },
                            "shadowOpacity": 0.23,
                            "shadowRadius": 2.62,
                            "width": 322.5,
                          }
                        }
                        testID={3}
                      >
                        <View>
                          <Image
                            source={1}
                            style={
                              Object {
                                "borderRadius": 8,
                                "height": "100%",
                                "resizeMode": "cover",
                                "width": "100%",
                              }
                            }
                          />
                        </View>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "rgba(255, 255, 255, 0.75)",
                              "borderRadius": 8,
                              "bottom": 0,
                              "flex": 1,
                              "height": "50%",
                              "justifyContent": "flex-end",
                              "position": "absolute",
                              "width": "100%",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "alignSelf": "center",
                                "color": "#D4A373",
                                "fontFamily": "MontserratBold",
                                "fontSize": 25,
                                "justifyContent": "flex-end",
                                "marginBottom": 10,
                              }
                            }
                          >
                            MASKS
                          </Text>
                        </View>
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "borderRadius": 8,
                            "elevation": 4,
                            "height": 466.9,
                            "margin": 10,
                            "opacity": 1,
                            "shadowColor": "#000",
                            "shadowOffset": Object {
                              "height": 2,
                              "width": 0,
                            },
                            "shadowOpacity": 0.23,
                            "shadowRadius": 2.62,
                            "width": 322.5,
                          }
                        }
                        testID={4}
                      >
                        <View>
                          <Image
                            source={1}
                            style={
                              Object {
                                "borderRadius": 8,
                                "height": "100%",
                                "resizeMode": "cover",
                                "width": "100%",
                              }
                            }
                          />
                        </View>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "rgba(255, 255, 255, 0.75)",
                              "borderRadius": 8,
                              "bottom": 0,
                              "flex": 1,
                              "height": "50%",
                              "justifyContent": "flex-end",
                              "position": "absolute",
                              "width": "100%",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "alignSelf": "center",
                                "color": "#D4A373",
                                "fontFamily": "MontserratBold",
                                "fontSize": 25,
                                "justifyContent": "flex-end",
                                "marginBottom": 10,
                              }
                            }
                          >
                            OILS
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      style={
                        Object {
                          "flexDirection": "row",
                        }
                      }
                    >
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "borderRadius": 8,
                            "elevation": 4,
                            "height": 466.9,
                            "margin": 10,
                            "opacity": 1,
                            "shadowColor": "#000",
                            "shadowOffset": Object {
                              "height": 2,
                              "width": 0,
                            },
                            "shadowOpacity": 0.23,
                            "shadowRadius": 2.62,
                            "width": 322.5,
                          }
                        }
                        testID={5}
                      >
                        <View>
                          <Image
                            source={1}
                            style={
                              Object {
                                "borderRadius": 8,
                                "height": "100%",
                                "resizeMode": "cover",
                                "width": "100%",
                              }
                            }
                          />
                        </View>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "rgba(255, 255, 255, 0.75)",
                              "borderRadius": 8,
                              "bottom": 0,
                              "flex": 1,
                              "height": "50%",
                              "justifyContent": "flex-end",
                              "position": "absolute",
                              "width": "100%",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "alignSelf": "center",
                                "color": "#D4A373",
                                "fontFamily": "MontserratBold",
                                "fontSize": 25,
                                "justifyContent": "flex-end",
                                "marginBottom": 10,
                              }
                            }
                          >
                            TOOLS
                          </Text>
                        </View>
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "borderRadius": 8,
                            "elevation": 4,
                            "height": 466.9,
                            "margin": 10,
                            "opacity": 1,
                            "shadowColor": "#000",
                            "shadowOffset": Object {
                              "height": 2,
                              "width": 0,
                            },
                            "shadowOpacity": 0.23,
                            "shadowRadius": 2.62,
                            "width": 322.5,
                          }
                        }
                        testID={6}
                      >
                        <View>
                          <Image
                            source={1}
                            style={
                              Object {
                                "borderRadius": 8,
                                "height": "100%",
                                "resizeMode": "cover",
                                "width": "100%",
                              }
                            }
                          />
                        </View>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "rgba(255, 255, 255, 0.75)",
                              "borderRadius": 8,
                              "bottom": 0,
                              "flex": 1,
                              "height": "50%",
                              "justifyContent": "flex-end",
                              "position": "absolute",
                              "width": "100%",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "alignSelf": "center",
                                "color": "#D4A373",
                                "fontFamily": "MontserratBold",
                                "fontSize": 25,
                                "justifyContent": "flex-end",
                                "marginBottom": 10,
                              }
                            }
                          >
                            SHAMPOOS
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </RCTScrollView>
            </View>
          </RNCSafeAreaView>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 1,
        "justifyContent": "center",
        "position": "absolute",
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "elevation": 4,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.23,
          "shadowRadius": 2.62,
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 5,
          }
        }
        testID="tabContainer"
      >
        <Text />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 5,
          }
        }
        testID="tabContainer"
      >
        <Text />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 5,
          }
        }
        testID="tabContainer"
      >
        <Text />
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
